File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/security/UserStoreKey.ts
module api.security {

    export class UserStoreKey implements api.Equitable {

        public static SYSTEM: UserStoreKey = new UserStoreKey('system');

        private id: string;

        constructor(id: string) {
            this.id = id;
        }

        toString(): string {
            return this.id;
        }

        equals(o: api.Equitable): boolean {
            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, UserStoreKey)) {
                return false;
            }

            var other = <UserStoreKey>o;
            if (!api.ObjectHelper.stringEquals(this.id, other.id)) {
                return false;
            }
            return true;
        }
    }
}

File: modules/wem-api/src/main/java/com/enonic/wem/api/security/UserStoreKey.java
package com.enonic.wem.api.security;

import org.apache.commons.lang.StringUtils;

import com.google.common.base.Preconditions;

public final class UserStoreKey
{
    private final static UserStoreKey SYSTEM = new UserStoreKey( "system" );

    private final String id;

    public UserStoreKey( final String id )
    {
        Preconditions.checkArgument( !StringUtils.isBlank( id ), "UserStoreKey cannot be blank: %s", id );
        this.id = id;
    }

    @Override
    public String toString()
    {
        return this.id;
    }

    @Override
    public boolean equals( final Object o )
    {
        if ( this == o )
        {
            return true;
        }
        if ( !( o instanceof UserStoreKey ) )
        {
            return false;
        }
        final UserStoreKey userStoreKey = (UserStoreKey) o;
        return id.equals( userStoreKey.id );
    }

    @Override
    public int hashCode()
    {
        return id.hashCode();
    }

    public static UserStoreKey system()
    {
        return SYSTEM;
    }
}
